pandas/tests/frame/test_constructors.py

```diff
@@ -300,18 +299,38 @@ def test_constructor_dtype_nocast_view_2d_array(self):
         df2 = DataFrame(df.values, dtype=df[0].dtype)
         assert df2._mgr.blocks[0].values.flags.c_contiguous
 
-    @pytest.mark.xfail(using_string_dtype() and HAS_PYARROW, reason="conversion copies")
-    def test_1d_object_array_does_not_copy(self):
+    def test_1d_object_array_does_not_copy(self, using_infer_string):
         # https://github.com/pandas-dev/pandas/issues/39272
         arr = np.array(["a", "b"], dtype="object")
         df = DataFrame(arr, copy=False)
+        if using_infer_string:
+            if df[0].dtype.storage == "pyarrow":
+                # object dtype strings are converted to arrow memory,
+                # no numpy arrays to compare
+                pass
+            else:
+                assert np.shares_memory(df[0].to_numpy(), arr)
+        else:
+            assert np.shares_memory(df.values, arr)
+
+        df = DataFrame(arr, dtype=object, copy=False)
         assert np.shares_memory(df.values, arr)
 
-    @pytest.mark.xfail(using_string_dtype(), reason="conversion copies")
-    def test_2d_object_array_does_not_copy(self):
+    def test_2d_object_array_does_not_copy(self, using_infer_string):
         # https://github.com/pandas-dev/pandas/issues/39272
         arr = np.array([["a", "b"], ["c", "d"]], dtype="object")
         df = DataFrame(arr, copy=False)
+        if using_infer_string:
+            if df[0].dtype.storage == "pyarrow":
+                # object dtype strings are converted to arrow memory,
+                # no numpy arrays to compare
+                pass
```

Comment:
Can you skip instead of passing? I think worth tracking in the future if there exists an equivalent Python function for Arrow, or alternatively if someone wants to take the time to write this out via ctypes

Reply:
> Can you skip instead of passing?

I still want to run the rest of the test, therefore was using a `pass`

> I think worth tracking in the future if there exists an equivalent Python function for Arrow, or alternatively if someone wants to take the time to write this out via ctypes

What do you mean exactly?

Reply:
> What do you mean exactly?

Even though we don't have a NumPy array in the case of pyarrow storage to compare, you could still create a small ctypes function that assures the memory locations are the same right? Or in the future maybe there will be a `pa.shares_memory(...)` function

Reply:
Our `tm.shares_memory` already does an attempt to check if two pyarrow array share the same memory, so this already somewhat exists. But, in this case we just can't test any shared memory, because the numpy object dtype array has been converted to a pyarrow array, which is never zero-copy.

Reply:
Ah ok thanks - missed the object part